### DISCUSSION TITLE
Adding Transitions for Theme changes

### DISCUSSION THREAD

**Discussion by arvind0598** [mantinedev/mantine#6476] [Q&A]
Hello, I have to support a bunch of different colors in my app, and the user can switch between them whenever they want to. 
This part is relatively straightforward, all I needed to do was to store the primary color in a context and update when needed.

The issue is that theme changes currently look abrupt because the transition properties don't include color changes in most places. I considered overriding `transition-property` but I'm not sure if doing something like this is the right thing to do, or if it might break other transitions already part of mantine.

```css
* {
  transition-property: all !important;
}
```

**Comment by rtivital** [comment]
And what is your question?

**Reply by arvind0598** [reply]
Is the wildcard transition-property override the right thing to do? Or can it break other transitions you've implemented?

**Reply by arvind0598** [reply]
As in, would the "right" solution be adding theme-able properties to each of the transition properties already defined in each of the classes? Or is just that wildcard thing enough?

**Reply by rtivital** [reply]
Yes, it will override all Mantine transition-property styles. If this is what you are going for in your application, then it's fine.
